src/engraving/layout/layoutchords.cpp

```diff
@@ -876,6 +878,7 @@ void LayoutChords::layoutChords3(const MStyle& style, std::vector<Note*>& notes,
         }
 
         Chord* chord = note->chord();
+        chords.insert(chord);
```

Comment:
Although this implementation is pretty elegant, I wonder whether it would be more efficient to collect chords at the side of the caller (where we must be iterating over chords anyway to collect the notes) and pass them here. I'm not sure whether that's really better and I guess you've already considered that option, so I'm just curious why you chose this solution.

Reply:
Nothing very deep, I just thought it would be a bit clumsy to be passing to this function both the vector of notes _and_ the vector of the chords containing them, cause anyway one always implies the other. Though you're almost certainly right that it would be more efficient, I'm gonna test it